examples/systemcalls/systemcalls.c

```diff
@@ -84,16 +130,59 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     // and may be removed
     command[count] = command[count];
 
-
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
-
-    va_end(args);
-
+    pid_t return_from_fork, result_from_wait;
+    int w_status;
+    
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if(fd == -1)
+    {
+    	perror("Error creating a file");
+	return false;
+    }
+    
+    return_from_fork = fork();
+	
+    if(return_from_fork == -1) 
+    	{
+		//Child process could not be created 
+		perror("fork"); 
+		return false;
+		     		
+    	} 
+    else if(return_from_fork == 0)
+    	{
+		//Child process created 
+		if (dup2(fd, 1) < 0) 
+		{ 
+			perror("dup2"); 
+			return false; 
+		}
+		close(fd);
+		//call execv() 	
+		execv(command[0],command);
+		perror("execvp"); 
+		exit(-1);	
+			
+    	}
+     else
+   	  {
+   	  	close(fd);
+    		result_from_wait = waitpid(return_from_fork,&w_status, 0);
+    		if(result_from_wait == -1)
+		{
+			//if error occurs 
+			syslog( LOG_ERR, "Error: %s", strerror(errno));
+			return false;
+		}
+		if(WIFEXITED(w_status))
+		{
+			if(w_status)
+				return false;
```

Comment:
Missing cleanup in these code paths.
Also see discussion in class about avoiding duplication between these two very similar functions.